### PR TITLE
PAL-614 Reservoir storage updates

### PR DESCRIPTION
**Pull request by ps2**
While digging into PAL-614, I did some refactoring to async, and added a test for lagging pump reconciliation. This PR doesn't change any functionality, but contains code improvements I think we should keep.

### CODE REVIEW

File: LoopKit Example/MasterViewController.swift [hunk 1]
```diff
@@ -237,10 +237,10 @@ class MasterViewController: UITableViewController {
 
                         unitVolume -= (drand48() * 2.0)
 
-                        group.enter()
-                        dataManager.doseStore.addReservoirValue(unitVolume, at: Date(timeIntervalSinceNow: index)) { (_, _, _, error) in
-                            group.leave()
-                        }
+//                        group.enter()
```
Comment (ps2) [inline LoopKit Example/MasterViewController.swift:240]: This is unused code; we should probably clean up the LoopKit example or redo it at some point.

File: LoopKit/InsulinKit/DoseStore.swift [hunk 2]
```diff
@@ -1184,6 +1174,8 @@ extension DoseStore {
             do {
                 var doses: [DoseEntry]
 
+                self.log.debug("here")
+
```
Comment (nhamming) [inline LoopKit/InsulinKit/DoseStore.swift:1178]: remove?

**Review by nhamming — APPROVED** [review]
LGTM

File: LoopKit/InsulinKit/DoseStore.swift (hunk 2 above)
Comment (ps2) [reply LoopKit/InsulinKit/DoseStore.swift:1178]: I've got it in my next pr, thanks.

**Pull request merged**